### DISCUSSION TITLE
Find filename under cursor with Telescope and fd?

### DISCUSSION THREAD

**Discussion by hirotaka** [LazyVim/LazyVim#2893] [Q&A] [answered]
I found something that looks close, which uses Vimscript.

https://www.reddit.com/r/neovim/comments/ook0o6/comment/h5zefzo/?utm_source=share&utm_medium=web3x&utm_name=web3xcss&utm_term=1&utm_content=share_button](https://www.reddit.com/r/neovim/comments/ook0o6/comment/h5zefzo/?utm_source=share&utm_medium=web3x&utm_name=web3xcss&utm_term=1&utm_content=share_button)

I'm trying to do this with the LazyVim config like the below:

```lua
return {
  {
    "telescope.nvim",
    keys = {
      {
         "<leader>fw",
         LazyVim.telescope("fd", { default_text = vim.fn.expand("<cword>") }),
         desc = "Find Files (Word)",
      },
    },
  },
}

```

However, I  couldn't pass the word on the cursor properly.
Any suggestion?

**Comment by dpetka2001** [comment, marked as answer]
LazyVim already has a keymap for this? `<leader>sw`

**Reply by hirotaka** [reply]
It's a bit different.
`<leader>sw` looks for a file that contains the word on the cursor.
What I want to do is to find the file name of the word on the cursor.

**Reply by dpetka2001** [reply]
```lua
      {
        "<leader>fw",
        function()
          return LazyVim.telescope("fd", { default_text = vim.fn.expand("<cword>") })()
        end,
        desc = "Find Files (Word)",
      },
```
This seems to do what you want.

**Reply by hirotaka** [reply]
It worked! Thanks!
